src/TheThingsNetwork.cpp

```diff
@@ -101,11 +100,18 @@ void TheThingsNetwork::onMessage(void (*cb)(const byte* payload, int length, int
   this->messageCallback = cb;
 }
 
-bool TheThingsNetwork::personalize(const String devAddr, const String nwkSKey, const String appSKey) {
+bool TheThingsNetwork::personalize(const char *devAddr, const char *nwkSKey, const char *appSKey) {
   reset();
-  sendCommand(F("mac set devaddr"), devAddr);
-  sendCommand(F("mac set nwkskey"), nwkSKey);
-  sendCommand(F("mac set appskey"), appSKey);
+  String addr = "mac set devaddr ";
+  String nKey = "mac set nwkskey ";
+  String aKey = "mac set appskey ";
+
+  addr.concat(devAddr);
+  nKey.concat(nwkSKey);
+  aKey.concat(appSKey);
+  sendCommand(addr);
+  sendCommand(nKey);
+  sendCommand(aKey);
```

Comment:
I'm sorry but I am not sure I understand what you mean... Is it a spacing problem like that: 
```
String eui = "mac set appeui ";
String key = "mac set appkey ";
key.concat(appKey);
eui.concat(appEui);                                                                                         

sendCommand(eui);
sendCommand(key);
return sendCommand(F("mac save")); 
 ``` 
or something else ?